### ISSUE TITLE
Question about current_score

### ISSUE THREAD

**Issue by yuanzhang22** [binli123/dsmil-wsi#25]
Hi,thank you for sharing this awesome project. 

I am trying to run dsmil on my own dateset. And I got 2 problems in train_tcga.py about current_score and class_prediction_bag.
1.I was wondering why current_score equals (sum(aucs) + avg_score + 1 - test_loss_bag)/4?
2.I was also confused about class_prediction_bag using the binary sigmoid prediction ((0.0*torch.sigmoid(max_prediction)+1.0*torch.sigmoid(bag_prediction)) for multiple classes(num_classes>1) instead of softmax prediction. And does this zero coefficient mean the max_prediction make no contribution to the final patient prediction? 

Many thanks!

**Comment by binli123** [comment]
1. This is just a criterion to select the best model which is the average score across AUCs, mean accuracy, and the loss value. You could change to only accuracy or AUCs to pick the best model.
2. Because the labels are created using distributed encoding, not one-hot encoding. e.g., class 1 will be labeled as [1, 0], class 2 will be [0 ,1], if both classes exist then [1, 1], negative class will be encoded as [0, 0]. The max_prediction has no contribution to the final score at inference after the model is trained. However, this is an adjustable parameter and the averaged score can be used.

**Comment by shimaxy** [comment]
Hi,

How about bag_prediction in attention_map.py? Do we need to run it through a sigmoid to get the scores? When I run attention_map as it is now, I get negative scores which result in skipping if loop of "bag_prediction[c] >= args.thres[c]" and getting error of "UnboundLocalError: local variable 'colored_tiles' referenced before assignment."

Thanks!

**Comment by binli123** [comment]
> Hi,
> 
> How about bag_prediction in attention_map.py? Do we need to run it through a sigmoid to get the scores? When I run attention_map as it is now, I get negative scores which result in skipping if loop of "bag_prediction[c] >= args.thres[c]" and getting error of "UnboundLocalError: local variable 'colored_tiles' referenced before assignment."
> 
> Thanks!

Good catch. But it is not only the reasons that cause the error. `torch.sigmoid` needs to be called since it is used in the training and testing loops so that the thresholds match. For the error, I need to re-arrange the lines to handle the negative detection results in order to avoid it. I have made these changes to the scripts.

**Comment by shimaxy** [comment]
True. There might be cases with negative detection results, and your fix will take care of that. In my case, passing through a sigmoid solved the issue.
Thanks for your quick response!
